modules/imgproc/src/shapedescr.cpp

```diff
@@ -492,6 +492,11 @@ static RotatedRect fitEllipseNoDirect( InputArray _points )
         CV_SWAP( box.size.width, box.size.height, tmp );
         box.angle = (float)(90 + rp[4]*180/CV_PI);
     }
+    else
+    {
+        box.angle = (float)(fmod(rp[4]*180/CV_PI,180.0));
```

Comment:
It would be nice to add small simple dedicated regression test for that (updated tests should catch removal of this code)

Reply:
Sure thing - will add as soon as (or if) I get the original tests working without hacks :')